test/Microsoft.DotNet.Tools.Tests.Utilities/TempFileSystem/FileNameUtilities.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DotNet.Tools.Test.Utilities
+{
+    /// <summary>
+    /// Implements a few file name utilities that are needed by the compiler.
+    /// In general the compiler is not supposed to understand the format of the paths.
+    /// In rare cases it needs to check if a string is a valid file name or change the extension 
+    /// (embedded resources, netmodules, output name).
+    /// The APIs are intentionally limited to cover just these rare cases. Do not add more APIs.
+    /// </summary>
+    internal static class FileNameUtilities
+    {
+        private const string DirectorySeparatorStr = "\\";
+        internal const char DirectorySeparatorChar = '\\';
+        internal const char AltDirectorySeparatorChar = '/';
+        internal const char VolumeSeparatorChar = ':';
```

Comment:
Talked offline. If this is part of a filename, there are bigger issues because `:` is also the path separator for Ubuntu.

Additionally we don't have any such files in our repo, where these tests will run.